src/main/java/org/spongepowered/api/ai/goal/agent/SwimmingGoal.java

```diff
@@ -22,18 +22,18 @@
  * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
  * THE SOFTWARE.
  */
-package org.spongepowered.api.entity.ai.task.builtin;
+package org.spongepowered.api.ai.goal.agent;
 
 import org.spongepowered.api.Sponge;
-import org.spongepowered.api.entity.ai.task.AITask;
-import org.spongepowered.api.entity.ai.task.AITaskBuilder;
+import org.spongepowered.api.ai.goal.Goal;
+import org.spongepowered.api.ai.goal.GoalBuilder;
 import org.spongepowered.api.entity.living.Agent;
 
-public interface SwimmingAITask extends AITask<Agent> {
+public interface SwimmingGoal extends Goal<Agent> {
```

Comment:
Since our `O` generic is the owner type, all subinterfaces need an `O` generic flag in the declaration. If it extended `Goal<AbMob>`, it will now be declared as `interface MyGoal<O extends AbMob> extends Goal<O>`